version: '3.8'
services:
 postgresql:
   image: postgres
   container_name: postgresql
   volumes:
     - postgresql-db:/var/lib/postgresql/data
   ports:
     - "5432:5432"
   env_file:
     - ./.env
   environment:
     POSTGRES_DB: ${DB_NAME}
     POSTGRES_USER: ${DB_USER}
     POSTGRES_PASSWORD: ${DB_PASSWORD}
   healthcheck:
     test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}"]
     interval: 10s
     timeout: 5s
     retries: 5

 django_gunicorn:
   build: ./
   volumes:
     - static:/app/project/static
     - media:/app/project/media
   env_file:
     - ./.env
   ports:
     - "8000:8000"
   depends_on:
     postgresql:
       condition: service_healthy

 nginx:
   build: 
    context: ./nginx
    dockerfile: Dockerfile.prod
   env_file:
     - ./.env
   environment:
     DOMAIN: ${PROD_ALLOWED_HOST}
   volumes:
     - static:/static
     - media:/media
     - certbot-web:/vol/www
     - certbot-certs:/etc/letsencrypt
     - nginx-ssl-conf:/vol/nginx_ssl_conf
   ports:
     - "80:80"
     - "443:443"
   depends_on:
     - django_gunicorn
 certbot:
   build: ./certbot
   environment:
     EMAIL: ${PROD_CERTBOT_EMAIL}
     DOMAIN: ${PROD_ALLOWED_HOST}
   volumes:
     - certbot-web:/vol/www
     - certbot-certs:/etc/letsencrypt/
   depends_on:
     - nginx

volumes:
 static:
 postgresql-db:
 certbot-web:
 certbot-certs:
 nginx-ssl-conf:
 media: